File: haskell/005.hs
-- No programming required here. Smallest num evenly divisible by 1..20:
-- 1) Get prime factors of 1 through 20
-- 2) Multiply them all together
-- 1*2*3*2*5*7*2*3*11*13*2*17*19 = (2^4)*(3^2)*5*7*11*13*17*19

solution = (2^4)*(3^2)*5*7*11*13*17*19

main = do print solution

File: c/005.c
#include <stdio.h>

int main(void)
{
    /* The smallest number divisible by all numbers 1-20 means that each
     * of those numbers will be contained in the factorization.
     * The prime numbers under 20 are 2, 3, 5, 7, 11, 13, 17, and 19.
     * The number with the most 2s in its prime factorization is 16: 2^4
     * The number with the most 3s in its prime factorization is 9: 3^2
     * No number has more than 1 of each of the other prime factors.
     * So, the number is just:
     */
    printf("%d\n", 2*2*2*2*3*3*5*7*11*13*17*19);
    return 0;
}
